### ISSUE TITLE
Class or interface $myVar does not exist for variable with inline doc comment with no type

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#292]
#291 seems to have the same root cause

I'm not 100% clear if the union type of `@var` is mandatory from the online docs, I've been assuming it is optional. http://docs.phpdoc.org/references/phpdoc/tags/var.html

Expected: no error
Observed: the below:

```
ERROR: UndefinedClass - src/inlinevar.php:7:5 - Class or interface $myVar does not exist
    $myVar->otherMethod();
```

```php
<?php

/** @param mixed $x */
function myvalue($x) : void {
    /** @var $myVar MyNS\OtherClass */
    $myVar = $x->conn()->method();
    $myVar->otherMethod();
}
```

**Comment by TysonAndre** [comment]
One possible quick fix (Is any variable other than $this valid in a union type?):

```
--- a/src/Psalm/Checker/FunctionLikeChecker.php
+++ b/src/Psalm/Checker/FunctionLikeChecker.php
@@ -1173,6 +1173,8 @@ abstract class FunctionLikeChecker extends SourceChecker implements StatementsSo
                 if ($return_type_token[0] === '$') {
                     if ($return_type === '$this') {
                         $return_type_token = 'static';
+                    } else {
+                        return '';
                     }
```

**Comment by muglug** [comment]
I think this is more of an `InvalidDocblock` issue than something Psalm should totally ignore (because the user probably meant `/** @var MyNS\OtherClass $myVar */`

**Comment by TysonAndre** [comment]
InvalidDocblock would be fine -- I wrote this before figuring out how to fix https://github.com/vimeo/psalm/pull/298

**Comment by muglug** [comment]
Fixed in https://github.com/vimeo/psalm/commit/b2ae9a2892a62ebfe56b2478f20e23b5cf7e12ed
